### DISCUSSION TITLE
Mehrfache Abfrage von Schalterstati über HTTP (Version 1.6.20)

### DISCUSSION THREAD

**Discussion by klaus-schauer** [camueller/SmartApplianceEnabler#292] [Installation und Betrieb]
Der Schalterstatus wird bei aktiviertem "Status über HTTP abfragen" teilweise bis zu vier mal sofort hintereinander abgefragt, obwohl ein valides Antwortpaket schon nach der ersten Abfrage versandt wurde. Wie kann man die Mehrfachabfragen unterbinden?

**Comment by camueller** [comment]
Wenn "Status über HTTP abfragen" aktiviert ist, wird das Ergebnis nicht gecacht, sondern jedes Mal abgefagt, wenn diese Information relevant ist. Wenn kein anderes Gerät den HTTP-Schalter schaltet, kannst Du "Status über HTTP abfragen" deaktivieren, dann wird der Status intern "gemerkt" und es erfolgen keine HTTP-Abfragen.

**Comment by klaus-schauer** [comment]
Danke für die schnelle Rückmeldung.

Das wäre ja auch völlig in Ordnung, wenn diese wiederholten Abfragen nicht direkt hintereinander also im ms-Takt erfolgen würden. Eine Abfrage in einem Intervall von mehreren zehn Sekunden ist ja durchaus sinnvoll.

Ich habe für Fhem eine HTTP API erstellt. Das klappt grundsätzlich sehr gut. Die mehrfachen gleichen http-Abfragen blockieren aber scheinbar das System kurzzeitig.

**Comment by camueller** [comment]
Ich hatte ganz vergessen zu erwähnen, das in SAE 2.0 die Abfrage periodisch alle 20s ausgeführt wird, weil der Status den anderen Komponenten via MQTT verfügbar gemacht wird. Das sollte Dein Problem lösen. Die 2.0er-Versionen sind zwar offiziell noch Beta-Versionen, aber ich setze sie seit Monaten auf meinen produktiven Raspis ein.

**Comment by klaus-schauer** [comment]
Danke für den Hinweis. Ich werde die neue Version verwenden.

Wenn ich die Dokumentation richtig gelesen habe, wird MQTT derzeit genutzt, um die Stati insbesondere in node-red zu visualisieren. Ist geplant MQTT, auch als weitere Schnittstelle zur Steuerung von Geräten und dem Auslesen von Stati zu verwenden?

**Comment by camueller** [comment]
Ja, es wird die Möglichkeit geben, die Zählerwerte neben GPIO, Modbus und HTTP auch via MQTT zu ermitteln. Gleichermassen wird es auch möglich sein, Geräte via MQTT zu schalten. Siehe https://github.com/camueller/SmartApplianceEnabler/issues/295
